src/layered_config_tree/main.py

```diff
@@ -346,33 +346,80 @@ def to_dict(self) -> dict[str, Any]:
+        does not exist *but the rest of the key path does*.
+
+        Returns
+        -------
+        The value at the key or nested keys. If the key path exists except for the
+        final value, the ``default_value`` is returned.
         """
-        return self[key] if key in self._children else default_value
+        if not isinstance(keys, (str, list)):
+            raise TypeError("The 'keys' parameter must be a string or a list of strings.")
 
-    def get_tree(self, key: str) -> LayeredConfigTree:
-        """Return the LayeredConfigTree at the key in the outermost layer of the config tree.
+        if isinstance(keys, str):
+            return self[keys] if keys in self._children else default_value
+        else:
+            # get the second-to-last value (which is by definition a LayeredConfigTree)
+            data = self.get_tree(keys[:-1])
+            return data[keys[-1]] if keys[-1] in data._children else default_value
+
+    def get_tree(self, keys: str | list[str]) -> LayeredConfigTree:
+        """Return the LayeredConfigTree at the key or key path from the outermost layer.
 
         Parameters
         ----------
-        key
-            The str we look up in the outermost layer of the config tree.
+        keys
+            The key or key path to look up from the outermost layer.
+
+        Returns
+        -------
+        The ``LayeredConfigTree`` located at the key or key path provided starting
+        from the outermost layer.
+
+        Raises
+        ------
+        TypeError
+            If the ``keys`` parameter is not a string or list of strings.
+        ConfigurationKeyError
+            If any of the keys in the key path do not exist in the tree.
+        ConfigurationError
+            If the data at the final key in the key path is not a
+            :class:`LayeredConfigTree`.
         """
-        data = self[key]
-        if isinstance(data, LayeredConfigTree):
-            return data
-        else:
+        if not isinstance(keys, (str, list)):
+            raise TypeError("The 'keys' parameter must be a string or a list of strings.")
+
+        if isinstance(keys, str):
+            keys = [keys]
+
+        data = self
+        for key in keys:
+            if key not in data:
+                raise ConfigurationKeyError(
+                    f"No value at key mapping '{keys[:keys.index(key) + 1]}'."
+                )
+            data = data[key]
+        if not isinstance(data, LayeredConfigTree):
             raise ConfigurationError(
-                f"The data you accessed using {key} with get_tree was of type {type(data)}, but get_tree must return a LayeredConfigTree."
+                f"The data you accessed using {keys} with get_tree was of type {type(data)}, "
+                "but get_tree must return a LayeredConfigTree."
             )
+        return data
 
     def get_from_layer(self, name: str, layer: str | None = None) -> Any:
```

Comment:
Not trying to cause scope creep, but should we extend the list or default value functionality to this method?

Reply:
create a ticket for this
- consolidate get from layer
- deprecate it
- keep existing get_from_layer logic (just move)
- add `layer` into get and get_tree